packages/components/src/input-control/types.ts

```diff
@@ -39,30 +54,78 @@ export type InputChangeCallback<
 > = ( nextValue: string | undefined, extra: { event: E } & P ) => void;
 
 export interface InputFieldProps extends BaseProps {
+	/**
+	 * Determines the drag axis.
+	 *
+	 * @default 'n'
+	 */
 	dragDirection?: DragDirection;
+	/**
+	 * If `isDragEnabled` is true, this controls the amount of `px` to have been dragged before
+	 * the drag gesture is actually triggered.
+	 *
+	 * @default 10
+	 */
 	dragThreshold?: number;
+	/**
+	 * If true, enables mouse drag gestures.
+	 *
+	 * @default false
+	 */
 	isDragEnabled?: boolean;
+	/**
+	 * If true, the `ENTER` key press is required in order to trigger an `onChange`.
+	 * If enabled, a change is also triggered when tabbing away (`onBlur`).
+	 *
+	 * @default false
+	 */
 	isPressEnterToChange?: boolean;
+	/**
+	 * A function that receives the value of the input.
+	 */
 	onChange?: InputChangeCallback;
 	onValidate?: (
 		nextValue: string,
 		event?: SyntheticEvent< HTMLInputElement >
 	) => void;
 	setIsFocused: ( isFocused: boolean ) => void;
 	stateReducer?: StateReducer;
+	/**
+	 * The current value of the input.
+	 */
 	value?: string;
 	onDragEnd?: ( dragProps: DragProps ) => void;
 	onDragStart?: ( dragProps: DragProps ) => void;
 	onDrag?: ( dragProps: DragProps ) => void;
```

Comment:
We're lacking in documentation for the drag stuff overall 😓 I'd say it's low priority though.

Reply:
Yeah, that's not an easy one because we want to be careful to document it in a way that doesn't necessarily expose the `@use-gesture/react` library as a dependency, in case we want to swap it in the future.

Reply:
Currently the generate type for these props is a bit convoluted (`(dragProps: Omit<FullGestureState<"drag">, "event"> & { event: unknown; }) => void`), do you think it's ok to keep it as-is, or should we try to simplify it at least for the docs description?

Reply:
Honestly I don't know what that "simplified" version would be 😅 I'm fine with keeping it as is for now, because it's probably irrelevant for 99% of consumers, at least without proper documentation.

Also TIL you can [override](https://storybook.js.org/docs/react/writing-docs/doc-block-argstable#customizing) type info in the args table. Nice option to have (when it's worth the maintenance cost)!